hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -182,4 +197,98 @@ object MergeOnReadSnapshotRelation {
     // when create PartitionedFile.
     path.toUri.toString
   }
+
+  /**
+   * Convert Filters to Catalyst Expressions and joined by And. If convert success return an
+   * Non-Empty Option[Expression],or else return None.
+   */
+  def convertToCatalystExpressions(filters: Array[Filter],
```

Comment:
can we encapsulate this conversion logic into its own class. I could see general use for this, beyond just partition pruning?

Reply:
Make sense to me!